docs/modeling/bounding-boxes.rst

```diff
@@ -13,137 +14,145 @@ image from a large survey catalog, each source should only be evaluated over the
-symmetric about the position that fully contains the model if the model has a
-finite extent. If a model does not have a finite extent, the choice for the
-``bounding_box_default`` limits is noted in the docstring. For example, see
-`astropy.modeling.functional_models.Gaussian2D.bounding_box_default`.
+For basic usage, see `Model.bounding_box
+<astropy.modeling.Model.bounding_box>`.  By default no
+`~astropy.modeling.Model.bounding_box` is set, except on model subclasses where
+a ``bounding_box`` property or method is explicity defined. The default is then
+the minimum rectangular region symmetric about the position that fully contains
+the model. If the model does not have a finite extent, the containment criteria
+are noted in the documentation. For example, see ``Gaussian2D.bounding_box``.
 
-The default function can also be set to any callable. This is particularly
-useful for fitting ``custom_model`` or ``CompoundModel`` instances.
+`Model.bounding_box <astropy.modeling.Model.bounding_box>` can be set by the
+user to any callable. This is particularly useful for fitting models created
+with `~astropy.modeling.custom_model` or as a compound model::
 
     >>> from astropy.modeling import custom_model
     >>> def ellipsoid(x, y, z, x0=0, y0=0, z0=0, a=2, b=3, c=4, amp=1):
-    ...     rsq = ((x-x0)/a) ** 2 + ((y-y0)/b) ** 2 + ((z-z0)/c) ** 2
+    ...     rsq = ((x - x0) / a) ** 2 + ((y - y0) / b) ** 2 + ((z - z0) / c) ** 2
     ...     val = (rsq < 1) * amp
     ...     return val
     ...
-    >>> def ellipsoid_bb(self):
-    ...     return ((self.z0 - self.c, self.z0 + self.c),
-    ...             (self.y0 - self.b, self.y0 + self.b),
-    ...             (self.x0 - self.a, self.x0 + self.a))
+    >>> class Ellipsoid3D(custom_model(ellipsoid)):
+    ...     # A 3D ellipsoid model
+    ...     @property
+    ...     def bounding_box(self):
+    ...         return ((self.z0 - self.c, self.z0 + self.c),
+    ...                 (self.y0 - self.b, self.y0 + self.b),
+    ...                 (self.x0 - self.a, self.x0 + self.a))
     ...
-    >>> Ellipsoid3D = custom_model(ellipsoid)
-    >>> Ellipsoid3D.bounding_box_default = ellipsoid_bb
     >>> model = Ellipsoid3D()
-    >>> model.bounding_box = 'auto'
     >>> model.bounding_box
     ((-4.0, 4.0), (-3.0, 3.0), (-2.0, 2.0))
 
-Efficient evaluation with ``render_model``
-------------------------------------------
+.. warning::
 
-When a model is evaluated over a range much larger than the model itself, it may
-be prudent to use `astropy.modeling.render_model` if efficiency is a concern.
-The ``render_model`` function can be used to evaluate a model on an array of the
-same dimensions. If no array is given, ``render_model`` will return a "postage
-stamp" array corresponding to the bounding box region. However, if
-``bounding_box`` is `None` an image or coordinates must be passed.
+    Note: Currently when creating a new compound model by combining multiple
```

Comment:
"Note:" is not necessary in a warning (and perhaps confusing, because ``.. note::`` means something different)